include/boost/gil/promote_integral.hpp

```diff
@@ -62,7 +59,7 @@ struct promote_to_larger
             typename promote_to_larger
                 <
                     T,
-                    typename boost::mpl::next<Iterator>::type,
+                    typename boost::mp11::mp_front<Iterator>::type,
```

Comment:
`mpl::next` is iterator over sequence, `mp11::mp_front` is element of sequence.
This `promote_to_larger` iteration needs to be re-worked a bit, using `mp_front` (or `mp_first` synonym) and `mp_at_c`, I think.

The mismatch above is the reason of the current CI failures.